### PR TITLE
[RLlib] Move to action distribution creation in RLMs to "Distribution.from_logits()"

### PR DESCRIPTION
**Pull request by ArturNiederfahrenhorst**
## Why are these changes needed?

Makes it so that RLMs use a method "Distribution.from_logits" that abstracts the particularities of the distribution from RL Modules such that they don't have to be aware of constructor kwargs etc.
RL Modules become mostly unaware of the particular distribution class but construct all distributions with
a class method `Distribution.from_logits(logits)`.

This is especially powerful since one functionality of the catalog is to communicate the requires input shape of the distribution to be the output shape of the policy head. With this PR, this shape becomes the shape of the logits and is abstracted from the forward methods.

### CODE REVIEW

**Review by kouroshHakha — APPROVED** [review]
One of the nicest PRs this week :) Just a few nits. Also test_io is failing. I think retry would just fix it.

File: rllib/models/tf/tf_distributions.py [hunk 1]
```diff
@@ -138,7 +138,7 @@ def _rsample(self, sample_shape=()):
 
     @classmethod
     @override(Distribution)
-    def from_logits(cls, logits: TensorType) -> "TfCategorical":
+    def from_logits(cls, logits: TensorType, **kwargs) -> "TfCategorical":
```
Comment (kouroshHakha) [inline rllib/models/tf/tf_distributions.py:141]: This should take in temperature, and use it, right?

**Review by kouroshHakha — APPROVED** [review]
